src/Interpolation1D.jl

```diff
@@ -65,44 +65,50 @@ Each column can have a different grid.
 """
 function interpolate1d!(
     ftarget::AbstractArray{FT, N},
-    xsource::AbstractArray{FT, N},
-    xtarget::AbstractArray{FT, N},
+    xsource::AbstractArray{FT, NSG},
+    xtarget::AbstractArray{FT, NTG},
     fsource::AbstractArray{FT, N},
     order,
     extrapolate = Flat(),
-) where {FT, N}
+) where {FT, N, NSG, NTG}
     @assert N ≥ 1
-    (nsource, coldims_source...) = size(xsource)
-    (ntarget, coldims_target...) = size(xtarget)
-    @assert coldims_source == coldims_target
-    @assert size(ftarget) == size(xtarget)
-    @assert size(fsource) == size(xsource)
+    @assert NSG == 1 || NSG == N # Source grid can be the same for all columns or different for each column
+    @assert NTG == 1 || NTG == N # Target grid can be the same for all columns or different for each column
+    (nsource, coldims_source...) = size(fsource)
+    (ntarget, coldims_target...) = size(ftarget)
+    @assert coldims_source == coldims_target # check if column count and shape is same between source and target
+    @assert size(ftarget, 1) == size(xtarget, 1) # check if column length is same between ftarget and xtarget
+    @assert size(fsource, 1) == size(xsource, 1) # check if column length is same between fsource and xsource
```

Comment:
We can avoid repeated calls to `size`
```suggestion
    @assert ntarget == size(xtarget, 1) # check if column length is same between ftarget and xtarget
    @assert nsource == size(xsource, 1) # check if column length is same between fsource and xsource
```